core/src/main/java/org/jdbi/v3/core/result/ResultBearing.java

```diff
@@ -224,19 +224,18 @@ default <T> ResultIterable<T> map(RowViewMapper<T> mapper) {
      * @see RowReducer
      */
     default <C, R> Stream<R> reduceRows(RowReducer<C, R> rowReducer) {
-        return scanResultSet((resultSetSupplier, ctx) -> {
-            try (ResultSet resultSet = resultSetSupplier.get()) {
+        return scanResultSet((resultSetSupplier, context) -> {
+            try (StatementContext ctx = context) {
+                ResultSet resultSet = resultSetSupplier.get();
                 RowView rowView = new RowViewImpl(resultSet, ctx);
 
                 C container = rowReducer.container();
                 while (resultSet.next()) {
                     rowReducer.accumulate(container, rowView);
                 }
-                return rowReducer.stream(container);
+                return rowReducer.stream(container).onClose(ctx::close);
```

Comment:
Isn't this `onClose(ctx::close)` redundant with the `try (ctx) { ... }` resource management? The context will be closed as if in a finally block here.